sqlmesh/core/table_diff.py

```diff
@@ -309,13 +323,33 @@ def name(e: exp.Expression) -> str:
             temp_table = exp.table_("diff", db="sqlmesh_temp", quoted=True)
 
             with self.adapter.temp_table(query, name=temp_table) as table:
-                summary_query = exp.select(
+                summary_sums = [
                     exp.func("SUM", "s_exists").as_("s_count"),
                     exp.func("SUM", "t_exists").as_("t_count"),
                     exp.func("SUM", "row_joined").as_("join_count"),
+                    exp.func("SUM", "null_grain").as_("null_grain_count"),
                     exp.func("SUM", "row_full_match").as_("full_match_count"),
                     *(exp.func("SUM", name(c)).as_(c.alias) for c in comparisons),
-                ).from_(table)
+                ]
+
+                if check_grain:
+                    distincts = [
+                        *(
+                            exp.func("COUNT", exp.func("DISTINCT", f"s__{c}")).as_(
+                                "distinct_count_s"
+                            )
+                            for c in index_cols
```

Comment:
why are you checking each distinct column, shouldn't it be the whole key? each column may not be unique on it's own, it's the combination of the index_cols that creates the unique key right?

Reply:
Yes that's my mistake, I adapted this now to check for distinct of the whole key instead